framework/src/modules/chain/logic/rounds_information.js

```diff
@@ -0,0 +1,152 @@
+const Bignumber = require('bignumber.js');
+const transactionTypes = require('../helpers/transaction_types.js');
+const Diff = require('../helpers/diff.js');
+
+const updateRoundInformationWithDelegatesForTransaction = function(
+	stateStore,
+	transaction,
+	forwardTick
+) {
+	if (transaction.type === transactionTypes.VOTE) {
+		(forwardTick
+			? transaction.asset.votes
+			: Diff.reverse(transaction.asset.votes)
+		)
+			.map(vote => {
+				// Fetch first character
+				const mode = vote[0];
+				const dependentId = vote.slice(1);
+				const balanceFactor = mode === '-' ? -1 : 1;
+				const account = stateStore.account.get(transaction.senderId);
+
+				const balance = new Bignumber(account.balance)
+					.multipliedBy(balanceFactor)
+					.toString();
+
+				const roundData = {
+					address: transaction.senderId,
+					delegatePublicKey: dependentId,
+					amount: balance,
+				};
+
+				return roundData;
+			})
+			.map(data => stateStore.round.add(data));
+	}
+};
+
+const updateSenderRoundInformationWithAmountForTransaction = function(
+	stateStore,
+	transaction,
+	forwardTick
+) {
+	const value = new Bignumber(transaction.fee).plus(transaction.amount);
+	const valueToUpdate = forwardTick
+		? value.multipliedBy(-1).toString()
+		: value.toString();
+	const account = stateStore.account.get(transaction.senderId);
+	let dependentPublicKeysToAdd = account.votedDelegatesPublicKeys || [];
+
+	if (transaction.type === transactionTypes.VOTE) {
+		const newVotes = forwardTick
+			? transaction.asset.votes
+			: Diff.reverse(transaction.asset.votes);
+		const downvotes = newVotes
+			.filter(vote => vote[0] === '-')
+			.map(vote => vote.slice(1));
+		const upvotes = newVotes
+			.filter(vote => vote[0] === '+')
+			.map(vote => vote.slice(1));
+		dependentPublicKeysToAdd = dependentPublicKeysToAdd.filter(
```

Comment:
maybe better to  name this variable what exactly is, like
```
const dependentPublicKeysWithoutUpvotes = ...
```
or maybe extract this function and name it `getDependentPublicKeysBeforeApply()`?